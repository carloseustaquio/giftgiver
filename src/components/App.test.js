import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import toJson from 'enzyme-to-json';

// render creates the snapchot obj! but
// doesn't have .simulate inside of it.
// mount and shallow doesn't creates the
// snapshot! should downgrade it? 

describe('App', () => { 
    const app = shallow(<App />)
    
    it("renders correctly", () => {
        expect(toJson(app)).toMatchSnapshot()
    })
    
    it("initializes with the `state` with an empty list of gifts", () => {
        expect(app.state().gifts).toEqual([])
    })

    describe("when clicking the `add-gift` button", () => {
        const id = 1
        
        beforeEach(() => {
            app.find(".btn-add").simulate('click')
        })
        
        afterEach(() => {
            app.setState({ gifts: [] })
        })
        
        it("adds a new gift to state", () => {
            expect(app.state().gifts).toEqual([{ id }])
        })
        
        it("adds a new gift to the render list", () => {
            expect(app.find(".gift-list").children().length).toEqual(1)
        })

        it("creates a gift component", () => {
            expect(app.find("Gift").exists()).toBe(true)
        })
        
        describe("when the user wants to remove the added gift", () => {
            beforeEach(() => {
                app.instance().removeGift(id)
            })

            it("removes the gift from state", () => {
                expect(app.state().gifts).toEqual([])
            })
        })
    })

})