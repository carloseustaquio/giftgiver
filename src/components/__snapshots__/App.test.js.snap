// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="container container-xs d-flex flex-column px-5 pb-4 position-relative"
>
  <h2
    className="mt-4 d-flex justify-content-between"
  >
    Gift Giver
    <Button
      active={false}
      className="btn-add"
      disabled={false}
      id="btn-add"
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Add Gift
    </Button>
  </h2>
  <div
    className="gift-list"
  />
</div>
`;
